sklearn/neighbors/_base.py

```diff
@@ -178,8 +177,9 @@ def _check_precomputed(X):
             "its handling of explicit zeros".format(graph.format)
         )
     copied = graph.format != "csr"
-    graph = check_array(graph, accept_sparse="csr")
-    check_non_negative(graph, whom="precomputed distance matrix.")
+    graph = check_array(
+        graph, accept_sparse="csr", ensure_non_negative=True, input_name="graph"
+    )
```

Comment:
why didn't you keep `precomputed distance matrix` ?

Reply:
Agreed, `graph` is an internal variable and people might get confused by the error message then.

Reply:
thanks @jeremiedbb , makes sense! I didn't notice there is more relevant naming there.